### PR TITLE
add composite action wrapper for golang-ci-workflow

### PR DESCRIPTION
**Pull request by pieterlukasse**
These changes move most of the steps in `.github/workflows/golang-ci-workflow.yaml` into a github action. The action is more generic and can be reused in other workflows in combination with "services" if needed.

Jira Ticket:  related to [PXP-9660](https://ctds-planx.atlassian.net/browse/PXP-9660)

### New Features

- composite action wrapper for golang-ci-workflow should allow us to reuse this workflow in combination with "services" if needed

### Breaking Changes

- removed the now obsolete `.github/workflows/golang-ci-workflow.yaml`

### CODE REVIEW

File: .github/workflows/golang-ci-workflow.yaml [hunk 1]
```diff
@@ -14,47 +14,11 @@ on:
...
-
-          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
-          # skip-build-cache: true
-      - name: Run Unit tests
-        run: |
-          go test ${{ inputs.TESTS_LOCATION }} -v -coverpkg=${{ inputs.TESTS_LOCATION }} -race -covermode atomic -coverprofile=./profile.cov
-      - name: Install goveralls
-        run: go install github.com/mattn/goveralls@latest
+      - name: Checkout code / lint code / install dependencies for goveralls
+        uses: uc-cdis/.github/.github/actions/golang-ci@master
```
Comment (pieterlukasse) [inline .github/workflows/golang-ci-workflow.yaml:21]: following [this example](https://stackoverflow.com/questions/59259783/how-to-combine-two-action-yml-files-in-one-repo), perhaps we could also write this as:
```
uses: ../actions/golang-ci
```

Comment (pieterlukasse) [reply .github/workflows/golang-ci-workflow.yaml:21]: it would make it more readable... 🤔

File: .github/actions/golang-ci/action.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        #           takes precedence over all other caching options.
+        skip-cache: true
+        # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+        # skip-pkg-cache: true
+        # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+        # skip-build-cache: true
+    - name: Run Unit tests
+      id: unit_tests_step
+      run: |
+        go test ${{ inputs.TESTS_LOCATION }} -v -coverpkg=${{ inputs.TESTS_LOCATION }} -race -covermode atomic -coverprofile="./profile.cov"
```
Comment (pieterlukasse) [inline .github/actions/golang-ci/action.yml:52]: "./profile.cov" is the same as teh value of `COVERAGE_OUTPUT_LOCATION`...but I couldn't find a way to specify this only once...tried referring to the   `outputs` part here with `${{ outputs...}}` , setting an environment variable, etc...all to no avail.

**Review by m0nhawk — DISMISSED** [review]

File: .github/actions/golang-ci/action.yml (hunk 2 above)
Comment (pieterlukasse) [reply .github/actions/golang-ci/action.yml:52]: that's what I had in https://github.com/uc-cdis/.github/pull/16/commits/aee5e30429c6a4c4cdb86d39701f981fe45971bf ... reverting.

**Review by m0nhawk — APPROVED** [review]

**Pull request merged**